spec/integration/recipes/lwrp_inline_resources_spec.rb

```diff
@@ -45,6 +45,33 @@ def action_a
     end
   end
 
+  context "with an inline resource with a property that shadows the enclosing provider's property" do
+    class LwrpShadowedPropertyTest < Chef::Resource::LWRPBase
+      PATH = ::File.join(Dir.tmpdir, "shadow-property.txt")
+      use_automatic_resource_name
+      actions :fiddle
+      property :content
+      action :fiddle do
+        file PATH do
+          content new_resource.content
+          action [:create, :delete]
+        end
+      end
+    end
+
+    after { File.delete(LwrpShadowedPropertyTest::PATH) rescue nil }
```

Comment:
This seems a potential way to bleed state around - why not let the exception get raised?